homeassistant/components/flipr/config_flow.py

```diff
@@ -0,0 +1,131 @@
+                errors = {}
+                errors["base"] = "cannot_connect"
+                return self._show_setup_form(errors)
+            except Exception as exception:  # pylint: disable=broad-except
+                errors = {}
+                errors["base"] = "unknown"
+                _LOGGER.exception(exception)
+                return self._show_setup_form(errors)
+
+            if len(flipr_ids) == 1:
+                self._flipr_id = flipr_ids[0]
+            else:
+                # No Flipr or multiple found : asks the user to enter it or select it.
+                self._possible_flipr_ids = flipr_ids
+                return await self.async_step_flipr_id()
+
+        # Check if already configured
+        await self.async_set_unique_id(self._flipr_id)
+        self._abort_if_unique_id_configured()
+
+        return self.async_create_entry(
+            title=self._flipr_id,
+            data={
+                CONF_EMAIL: self._username,
+                CONF_PASSWORD: self._password,
+                CONF_FLIPR_ID: self._flipr_id,
+            },
+        )
+
+    def _show_setup_form(self, errors=None):
+        """Show the setup form to the user."""
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_EMAIL): str, vol.Required(CONF_PASSWORD): str}
+            ),
+            errors=errors,
+        )
+
+    async def _authenticate_and_search_flipr(self) -> List[str]:
+        """Validate the username and password provided and searches for a flipr id."""
+        client = await self.hass.async_add_executor_job(
+            FliprAPIRestClient, self._username, self._password
+        )
+
+        flipr_ids = await self.hass.async_add_executor_job(client.search_flipr_ids)
+
+        return flipr_ids
+
+    async def async_step_flipr_id(self, user_input=None):
+        """Handle the initial step."""
+        if not user_input:
+            # Creation of a select with the proposal of flipr ids values found by API.
+            if len(self._possible_flipr_ids) > 1:
+                flipr_ids_for_form = {}
+                for flipr_id in self._possible_flipr_ids:
+                    flipr_ids_for_form[flipr_id] = f"{flipr_id}"
+
+                return self.async_show_form(
+                    step_id="flipr_id",
+                    data_schema=vol.Schema(
+                        {
+                            vol.Required(CONF_FLIPR_ID): vol.All(
+                                vol.Coerce(str), vol.In(flipr_ids_for_form)
+                            )
+                        }
+                    ),
+                )
+            # Else : no flipr_id found. Allow the user to enter it.
+            # It may cause no data found afterwards in metrics...
+            return self.async_show_form(
+                step_id="flipr_id",
+                data_schema=vol.Schema({vol.Required(CONF_FLIPR_ID): str}),
```

Comment:
It seems that flipr id's always need to be 9 characters? Would it make sense to validate this? Or even better, are you able to validate via the API if a flipr id is correct and if it will return data?

I don't know enough about the device / API to give a good advice, but the method seem very error prone. Does it happen often that no flipr id's are returned when you are logged in?

Reply:
No, there is no rule on flipr id size from my knowledge.
And there is no API to valide a flipr_id.
I imagined this in the case the user would use homeassistant's flipr integration and he has not initialized or configured it  through the official mobile app. This should be a very rare case so after rethinking of it, it is certainly better to remove this case from the config setup and explain it through an error message.
This is what I did : I changed it to an error case during setup.

Reply:
It would make sense to make the Config Flow as simple as possible. If someone has multiple Flipr devices linked to one account, do you need to create a config entry for every device (by Flipr Id)?

Best would be to just pull all the linked Flipr ID's on load / reload and create devices + entities based on the available devices.

Reply:
@cnico wouldn't it make sense to remove the logic where you only select a single Flipr ID? Doesn't it make sense to do this in __init__ thus that all devices linked to your account will be added? Are there any usecases where you only want to select a single flipr id from your account?

```client.search_flipr_ids```

Reply:
Yes, you are right, if the user has many fliprs, which is not my case for now, the best thing is probably to directly add all entities. If some are useless, it should not be a problem.
I will refactor that way and push it.